### ISSUE TITLE
Gogs with PV does not start

### ISSUE THREAD

**Issue by chbe8475** [fabric8io/gofabric8#204]
I installed using gofabric8 version 0.4.72 on a one node kubernetes cluster running on Ubuntu 14.04.
I set the --pv flag to true.
Gogs does not start up.
`kubectl logs gogs-xyx-123` gives me:

```
mkdir: cannot create directory '/app/gogs/data/git': Permission denied
mkdir: cannot create directory '/app/gogs/data/ssh/': Permission denied
/opt/gogs/start.sh: line 24: cd: /app/gogs/data/ssh/: No such file or directory
panic: open gogs.rsa: permission denied

goroutine 1 [running]:
panic(0x519120, 0xc4202a9290)
    /usr/local/go/src/runtime/panic.go:500 +0x1a1
main.main()
    /home/jdyson/go/src/github.com/fabric8io/docker-gogs/cmds/ssh-hostkeygen/ssh-hostkeygen.go:41 +0x2c1
2016/09/27 10:24:58 Creating admin user:  true
2016/09/27 10:24:58 [T] Custom path: /opt/gogs/custom
2016/09/27 10:24:58 [T] Log path: /app/gogs/data/logs
2016/09/27 10:24:58 [I] Gogs: Go Git Service 
2016/09/27 10:24:58 [.../gogs/models/repo.go:121 NewRepoContext()] [E] Fail to set git user.email(exit status 255): error: could not lock config file /app/gogs/data/.gitconfig: Permission denied
```

I worked around this by logging in to the ubuntu server running kubernetes and changed the permission on the /data/gogs-data to 777. This is not optimal but it works for now :-)

**Comment by rawlingsj** [comment]
@chbe8475 currently this will only work on minikube and minishift, sorry we ought to highlight that.  We're looking to add initcontrainer annotations that chmod the filesystem like this is you want to try..
```
        pod.alpha.kubernetes.io/init-containers: '[
            {
                "name": "init",
                "image": "busybox",
                "command": ["chmod", "777", "/gogs-data"],
                "volumeMounts": [
                    {
                        "name": "gogs-data",
                        "mountPath": "/gogs-data"
                    }
                ]
            }
        ]'
```

**Comment by chbe8475** [comment]
Thanks @rawlingsj I am just happy that I got it working on kubernetes even though I had to do some manual stuff. I have created projects in gogs and in jenkins and then deleted the pods. When the pods were running again I still had my projects so as I said, I am happy.

I guess I will try your suggestion next time I install. Which probably will be sooner than later.

**Comment by rawlingsj** [comment]
Ok sounds good.  

FWIW we're planning on automating the initcontainer approach.  The aim is that things will just work OOTB.  I think we should keep this issue open to track adding initcontainer to set the mounted PV permissions.

**Comment by rawlingsj** [comment]
initcontainer approach to chmod PV on Kubernetes has been implemented now see fabric8io/fabric8-maven-plugin#533
